### DISCUSSION TITLE
Loading bar when doing computation on the backend?

### DISCUSSION THREAD

**Discussion by 0ctopirate** [tauri-apps/tauri#4069] [Ideas]
When trying to do large computation on the backend, it would be nice to display the status of the loop, but I'm not sure how you'd go about this. Maybe you could try to use [generators](https://doc.rust-lang.org/stable/std/ops/trait.Generator.html)?

Also, when the program is alt-tabbed and the backend is doing computation, you can't tab back into it. Is there any way to fix this?

**Comment by FabianLars** [comment]
You could emit events from inside your commands to report the current progress. Here's a slightly more complicated example (because it uses streams) from one of our plugins: <https://github.com/tauri-apps/tauri-plugin-upload/blob/dev/src/lib.rs#L73>

> Also, when the program is alt-tabbed and the backend is doing computation, you can't tab back into it. Is there any way to fix this?

You can try making your command `async` so that it doesn't run on the main thread.
